cookbooks/bcpc/recipes/powerdns.rb

```diff
@@ -45,259 +44,214 @@
 subnet = node[:bcpc][:management][:subnet]
 raise "Did not get a subnet" if not subnet
 
-%w{pdns-server pdns-backend-mysql}.each do |pkg|
-    package pkg do
-        action :upgrade
-    end
-end
+if node[:bcpc][:management][:vip] and get_nodes_for("mysql").length() > 0
+#  make_config('mysql-pdns-user', "pdns")
+#  make_config('mysql-pdns-password', secure_password)
+
+  allnodes = Array.new
+  get_all_nodes.each do |nodeobj| 
+    tempHash = Hash.new 
+    tempHash['hostname'] = nodeobj.hostname 
+    tempHash['management_ip'] = nodeobj.bcpc.management.ip 
+    tempHash['storage_ip'] = nodeobj.bcpc.storage.ip
+    tempHash['floating_ip'] = nodeobj.bcpc.floating.ip
+    allnodes.push(tempHash)
+  end
+  
+  if !node[:bcpc][:management][:viphost].nil?
+    tempHash = Hash.new
+    tempHash['hostname'] = node[:bcpc][:management][:viphost].split('.')[0]
+    tempHash['management_ip'] = node[:bcpc][:management][:vip]
+    tempHash['floating_ip'] =  node[:bcpc][:floating][:vip]
+    tempHash['storage_ip'] =  node[:bcpc][:management][:vip]
+    allnodes.push(tempHash)
+  end
+
+  node.set['pdns']['authoritative']['package']['backends'] = ['gmysql']
```

Comment:
`node.set` is deprecated; please use `node.<level>` (e.g. level being `default`, or `override`).